rescheduler.go

```diff
@@ -377,36 +335,32 @@ func createEventRecorder(client kube_client.Interface) kube_record.EventRecorder
 // scheduled on the node, and returns the node if it finds a suitable one.
 // Currently sorts nodes by most requested CPU in an attempt to fill fuller
 // nodes first (Attempting to bin pack)
-func findSpotNodeForPod(predicateChecker *simulator.PredicateChecker, nodeInfos []*nodes.NodeInfo, pod *apiv1.Pod) *nodes.NodeInfo {
-	for _, nodeInfo := range nodeInfos {
-		kubeNodeInfo := schedulercache.NewNodeInfo(nodeInfo.Pods...)
-		kubeNodeInfo.SetNode(nodeInfo.Node)
-
+func findSpotNodeForPod(predicateChecker simulator.PredicateChecker, spotSnapshot simulator.ClusterSnapshot, nodes nodes.NodeInfoArray, pod *apiv1.Pod) string {
```

Comment:
Could you change this to use the idiom of Go (string, error) with the return value ?
Right now you are hiding the error from the logs.

Reply:
Discussed offline, I'm happy to modify code to be more in line with Golang idioms. 

In this specific case though, those errors are not exceptions, they are basically a negative answer. We pass a lot of pod and node information to the simulator and basically ask if it is even possible for a specific node to host an extra pod given the current state. If so, err is nil and we got the answer we were looking for, we return that node name. If the answer is no, err will have the reason as to why, but we don't really care. We don't even check that error, but we don't stop either. We just check the next node. And if none can accomodate a pod, it means that pod can't be rescheduled, we return an empty string. Returning the last "error" at that point would just have an explanation for the last node refusing the pod, not the full story. And the full story is not really useful either.

`I1007 14:10:03.457478   70978 rescheduler.go:272] Cannot drain node: pod queue-throttling-service/queue-throttling-service-deploy-98b5d88bd-45xtx can't be rescheduled on any existing spot node
`
Let me know what you think.

Reply:
I like the full story, maybe someone else has a different view ?